### PR TITLE
adds eslint, removes jshint and jscs, also some code style fixes

### PR DESCRIPTION
**Pull request by zuker**
Well-established rules are marked as errors in `.eslintrc.js`, warnings - for discussion.
BTW @bolgovr, as far as I remember, you've mentioned jscs's `--fix` cli argument, eslint has the same.

Here's list of rules for discussion:
##### `quote-props` - consistent-as-needed

``` javascript
/*eslint quote-props: ["error", "consistent-as-needed"]*/

var object1 = {
    'foo': 'bar',
    'baz': 42,
    'qux-lorem': true
};

var object2 = {
    foo: 'bar',
    baz: 42
};
```
##### `new-cap` - default

``` javascript
/*eslint new-cap: ["error", { "newIsCap": true }]*/

var friend = new Person();
```
##### `object-curly-spacing` - always

``` javascript
/*eslint object-curly-spacing: ["error", "always"]*/

var obj = {};
var obj = { 'foo': 'bar' };
var obj = { 'foo': { 'bar': 'baz' }, 'qux': 'quxx' };
var obj = {
  'foo': 'bar'
};
var { x } = y;
import { foo } from 'bar';
```
##### `object-curly-newline` - multiline (`minProperties: 5`)

``` javascript
/*eslint object-curly-newline: ["error", { "ObjectExpression": "always", "ObjectPattern": { "minProperties": 5 } }]*/
/*eslint-env es6*/

let a = {};
let b = {
    foo: 1
};
let c = {
    foo: 1, 
    bar: 2, 
    baz: 3, 
    qux: 4, 
    foobar: 5
};
let d = {
    foo: 1,
    bar: 2
};
let e = {
   foo: function() {
      dosomething();
   }
};

let {} = obj;
let {f} = obj;
let {
    g,
    h, 
    i, 
    j, 
    k
} = obj;
let {
    i,
    j
} = obj;
let {
  k = function() {
      dosomething();
  }
} = obj;
```
##### `strict: global` - http://eslint.org/docs/rules/strict#global (I think this should be just fixed everywhere :)

``` javascript
/*eslint strict: ["error", "global"]*/

'use strict';

function foo() {
}
```
##### `no-use-before-define`

``` javascript
/*eslint no-use-before-define: "error"*/
/*eslint-env es6*/

var a;
a = 10;
alert(a);

function f() {}
f(1);

var b = 1;
function g() {
    return b;
}

// With blockBindings: true
{
    let C;
    c++;
}
```
##### `arrow-body-style` - http://eslint.org/docs/rules/arrow-body-style#as-needed

``` javascript
/*eslint arrow-body-style: ["error", "as-needed"]*/
/*eslint-env es6*/

let foo = () => 0;
let foo = (retv, name) => {
    retv[name] = true;
    return retv;
};
let foo = () => ({
    bar: {
        foo: 1,
        bar: 2,
    }
});
let foo = () => { bar(); };
let foo = () => {};
let foo = () => { /* do nothing */ };
let foo = () => {
    // do nothing.
};
let foo = () => ({ bar: 0 });
```
##### object-property-newline

``` javascript
/*eslint object-property-newline: "error"*/

var obj = {
    foo: "foo",
    bar: "bar",
    baz: "baz"
};
```

### CODE REVIEW

**Comment by pnedelko** [comment]
@zuker are these rules similar to the previous ones? Currently almost all our projects obey `jscs` rules, would be nice not to do a lot of changes in code style.

**Comment by zuker** [comment]
@pnedelko all rules are converted from jscs/jshint to eslint. Rules from discussion list are either not well established or need to be fixed (i.e. `'use strict';`)

**Comment by zuker** [comment]
Also we can always set rule severity to `warning` to be aware of it, but style check will not fail.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        'eqeqeq': ERROR,
+        'no-use-before-define': WARN,
+        'no-undef': ERROR,
+        'no-unused-vars': WARN,
+        'no-mixed-spaces-and-tabs': ERROR,
+        'operator-linebreak': [
+            ERROR,
+            'before'
+        ],
+        'no-console': [
```
Comment (bolgovr) [inline .eslintrc.js:137]: I think`'no-console': [WARN]` would be enough

**Pull request merged**